lib/oauth_middleware.js

```diff
@@ -141,10 +141,10 @@ module.exports = function oauth(options) {
   return function (req, res, next) {
     if (req.url.indexOf(options.callbackPath) === 0) {
       oauthCallback(req, res, next, options);
-    } else if (req.url.indexOf(options.loginPath) === 0) {
-      login(req, res, next, options);
     } else if (req.url.indexOf(options.logoutPath) === 0) {
       logout(req, res, next, options);
+    } else if (req.url.indexOf(options.loginPath) === 0) {
```

Comment:
@fengmk2 using the original code, when you visit '/oauth/logout', login() gets called.